compiler/src/iree/compiler/Dialect/LinalgExt/Utils/Utils.cpp

```diff
@@ -548,29 +551,53 @@ getIGEMMGenericConvDetails(linalg::LinalgOp linalgOp) {
 
   SmallVector<AffineExpr> dims(numParallelDims + numKDims);
   bindDimsList<AffineExpr>(ctx, dims);
+  // Build the result map with dims in canonical order.
   auto resultMap = AffineMap::get(
       numParallelDims + numKDims, 0,
       SmallVector<AffineExpr>(dims.begin(), dims.begin() + numParallelDims),
       ctx);
 
+  // Build a mapping from canonical dim positions to their positions in the
+  // original output map.
+  SmallVector<unsigned> canonicalToOriginalDimPos;
+  ArrayRef<AffineExpr> originalExprs = outputMap.getResults();
+  ArrayRef<AffineExpr> canonicalExprs = resultMap.getResults();
+  for (AffineExpr canonicalExpr : canonicalExprs) {
+    unsigned canonicalPos = cast<AffineDimExpr>(canonicalExpr).getPosition();
+    auto it = llvm::find_if(originalExprs, [&](AffineExpr e) {
+      return cast<AffineDimExpr>(e).getPosition() == canonicalPos;
+    });
+    assert(it != originalExprs.end() && "dim not found in originalExprs");
+    canonicalToOriginalDimPos.push_back(
+        std::distance(originalExprs.begin(), it));
+  }
```

Comment:
I think this is invert permutation of "outputMap.getResults()..getPostion()" . Also I think there is only one test exercising this logic `conv_1d_ncw_fcw_transpose_maps` and in that one I think in that one the positions are [0,2,1] and the invert permutation of that is again [0,2,1] so we should add a test where that is not the case g.g if output map was [1,2,0] then invert permutations should be [2,0,1]

Reply:
Thanks, I've refactored the code to use `invertPermutationVector`.

Regarding to the test, there's an existing one that can test the logic https://github.com/iree-org/iree/blob/ed24953a8228b5e198a3db8af893e1a53209e0bd/compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/test/conv_to_im2col.mlir#L148
The original is [0, 2, 3, 1], and the inverse is [0, 3, 1, 2].

Reply:
Nice!